ansible/roles/dhcpd/defaults/main.yml

```diff
@@ -1,274 +1,314 @@
-                           if (ansible_default_ipv4.gateway|d())
-                           else [] }}' ]
+# Additional options to start the dhcpd processes with.
+# See https://manpages.debian.org/dhcpd.8
+dhcpd__options: ''
 
                                                                    # ]]]
-# .. envvar:: dhcpd_relay_interfaces [[[
+# .. envvar:: dhcpd__interfacesv4 [[[
 #
-# List of network interfaces that dhcrelay should listen on
-dhcpd_relay_interfaces: []
+# List of network interfaces to serve DHCPv4 requests on.
+dhcpd__interfacesv4: [ '{{ ansible_local.ifupdown.external_interface
+                           if ansible_local.ifupdown.external_interface|d()
+                           else ansible_default_ipv4.interface }}' ]
 
                                                                    # ]]]
-# .. envvar:: dhcpd_relay_options [[[
+# .. envvar:: dhcpd__interfacesv6 [[[
 #
-# Additional dhcrelay options
-dhcpd_relay_options: '{{ "-" + dhcpd_ipversion }}'
+# List of network interfaces to serve DHCPv6 requests on.
+dhcpd__interfacesv6: '{{ dhcpd__interfacesv4
+                         if ansible_default_ipv6.address|d()
+                         else [] }}'
                                                                    # ]]]
                                                                    # ]]]
-# ISC DHCP Server configuration [[[
-# ---------------------------------
+# DHCP server configuration [[[
+# -----------------------------
 
-# .. envvar:: dhcpd_server_options [[[
+# .. envvar:: dhcpd__authoritative [[[
 #
-# dhcpd(8) options
-dhcpd_server_options: '{{ "-" + dhcpd_ipversion }}'
-                                                                   # ]]]
-                                                                   # ]]]
-# DHCP main configuration [[[
-# ---------------------------
-
-# .. envvar:: dhcpd_authoritative [[[
-#
-# Is this DHCP server authoritative?
-dhcpd_authoritative: False
+# Whether to serve authoritative responses.
+dhcpd__authoritative: False
 
                                                                    # ]]]
-# .. envvar:: dhcpd_log_facility [[[
+# .. envvar:: dhcpd__log_facility [[[
 #
-# Log facility to use
-dhcpd_log_facility: 'local7'
+# Syslog facility to use.
+# See https://manpages.debian.org/syslog.3#Values_for_facility
+dhcpd__log_facility: 'daemon'
 
                                                                    # ]]]
-# .. envvar:: dhcpd_interfaces [[[
+# .. envvar:: dhcpd__default_lease_time [[[
 #
-# List of network interfaces to listen on for DHCP requests
-# If this list is empty, Ansible will try to guess correct interfaces
-# automatically
-dhcpd_interfaces: []
+# The length in seconds that will be assigned to a lease if the client
+# requesting the lease does not ask for a specific expiration time. This is used
+# for both DHCPv4 and DHCPv6 leases (it is also known as the "valid lifetime" in
+# DHCPv6).
+dhcpd__default_lease_time: '43200'
```

Comment:
Is this 12 hours? Perhaps you could write it as:
```yaml
'{{ 60 * 60 * 12 }}'
```
to be easier to understand by a reviewer. The same with max lease time.